### PR TITLE
Replace distance with height for pre images

### LINKED ISSUE #320: Agora replace distance with height for pre images
https://github.com/bosagora/agora/pull/1902

### PR DESCRIPTION
**Pull request by TrustHenry**
This work was synchronized due to the change of Agora.
This changes the database field distance to height in the validators table as it now is the block height for the pre-image.

Related to #320

### CODE REVIEW

File: src/Stoa.ts [hunk 1]
```diff
@@ -294,38 +296,41 @@ class Stoa extends WebService
...
                         }
                         else
                         {
-                            preimage_distance = NaN;
+                            preimage_height_str = "";
                             result_preimage_hash = new Hash(Buffer.alloc(Hash.Width));
                         }
                     }
-
+
```
Comment (MichaelKim20) [inline src/Stoa.ts]: I don't think this is necessary.

File: src/modules/storage/LedgerStorage.ts [hunk 2]
```diff
@@ -1241,16 +1241,14 @@ export class LedgerStorage extends Storages
              WHERE avail_height <= ` + cur_height + `
                AND ` + cur_height + ` < (avail_height + cycle_length)
              GROUP BY utxo_key) as enrollments
-        INNER JOIN utxos
-            ON enrollments.utxo_key = utxos.utxo_key
```
Comment (MichaelKim20) [inline src/modules/storage/LedgerStorage.ts:1245]: This part has been modified.
Good

File: tests/Utils.ts [hunk 3]
```diff
@@ -57,14 +57,14 @@ export const sample_preImageInfo =
...
-        "distance": 6
+        "height": "6"
     };
 
 export const sample_reEnroll_preImageInfo =
     {
         "utxo": "0x2f8b231aa4fd35c6a5c68a97fed32120da48cf6d40ccffc93d8dc41a3016eb56434b2c44144a38efe459f98ddc2660b168f1c92a48fe65711173385fb4a269e1",
         "hash": "0xe51e1cc8dfdcdcd02586c9648c6977504eade2dffc8f3289e7ae7e501c2879f99af6a199ccad499be02a66d409ca4ab51b35f1c3a06a82464ce4efcfeb3ade33",
-        "distance": 6
+        "height": "12"
```
Comment (MichaelKim20) [inline tests/Utils.ts:67]: Nice catch.

**Review by MichaelKim20 — APPROVED** [review]

File: src/Stoa.ts (hunk 1 above)
Comment (TrustHenry) [reply src/Stoa.ts]: I fixed it. Thanks.

**Comment by TrustHenry** [comment]
@MichaelKim20 Thanks for the review.

**Pull request merged**